lib/src/model/robotoff_nutrient_extraction.dart

```diff
@@ -0,0 +1,201 @@
+    this.count,
+    this.insights,
+  });
+
+  NutrientExtractionInsight? get getLatestInsights {
+    insights?.sort((a, b) => a.completedAt!.compareTo(b.completedAt!));
+    return insights?.last;
+  }
+
+  NutrientEntity? getNutrientEntity(String nutrientKey) {
+    return getLatestInsights?.data?.nutrients?[nutrientKey];
+  }
+
+  factory RobotoffNutrientExtractionResult.fromJson(
+          Map<String, dynamic> json) =>
+      _$RobotoffNutrientExtractionResultFromJson(json);
+
+  @override
+  Map<String, dynamic> toJson() =>
+      _$RobotoffNutrientExtractionResultToJson(this);
+}
+
+@JsonSerializable()
+class NutrientEntity {
+  final int? start;
+  final int? end;
+  final String? text;
+  final String? unit;
+  final double? score;
+  final bool? valid;
+  final String? value;
+  final String? entity;
+  @JsonKey(name: 'char_start')
+  final int? charStart;
+  @JsonKey(name: 'char_end')
+  final int? charEnd;
+
+  const NutrientEntity({
+    this.start,
+    this.end,
+    this.text,
+    this.unit,
+    this.score,
+    this.valid,
+    this.value,
+    this.entity,
+    this.charStart,
+    this.charEnd,
+  });
+
+  factory NutrientEntity.fromJson(Map<String, dynamic> json) =>
+      _$NutrientEntityFromJson(json);
+
+  Map<String, dynamic> toJson() => _$NutrientEntityToJson(this);
+}
+
+@JsonSerializable()
+class NutrientAnnotationData {
+  final String? unit;
+  final String? value;
+
+  const NutrientAnnotationData({
+    this.unit,
+    this.value,
+  });
+
+  factory NutrientAnnotationData.fromJson(Map<String, dynamic> json) =>
+      _$NutrientAnnotationDataFromJson(json);
+
+  Map<String, dynamic> toJson() => _$NutrientAnnotationDataToJson(this);
+}
+
+@JsonSerializable()
+class NutrientAnnotation {
+  final Map<String, NutrientAnnotationData>? nutrients;
+  @JsonKey(name: 'serving_size')
+  final String? servingSize;
+  @JsonKey(name: 'nutrition_data_per')
+  final String? nutritionDataPer;
+
+  const NutrientAnnotation({
+    this.nutrients,
+    this.servingSize,
+    this.nutritionDataPer,
+  });
+
+  factory NutrientAnnotation.fromJson(Map<String, dynamic> json) =>
+      _$NutrientAnnotationFromJson(json);
+
+  Map<String, dynamic> toJson() => _$NutrientAnnotationToJson(this);
+}
+
+@JsonSerializable()
+class NutrientDataWrapper {
+  final Map<String, List<NutrientEntity>>? entities;
+  final Map<String, NutrientEntity>? nutrients;
+  final NutrientAnnotation? annotation;
+  @JsonKey(name: 'was_updated')
+  final bool? wasUpdated;
+
+  const NutrientDataWrapper({
+    this.entities,
+    this.nutrients,
+    this.annotation,
+    this.wasUpdated,
+  });
+
+  factory NutrientDataWrapper.fromJson(Map<String, dynamic> json) =>
+      _$NutrientDataWrapperFromJson(json);
+
+  Map<String, dynamic> toJson() => _$NutrientDataWrapperToJson(this);
+}
+
+@JsonSerializable()
+class NutrientExtractionInsight extends JsonObject {
+  @JsonKey(name: 'id')
+  final String? insightId;
+  final String? barcode;
+  final String? type;
```

Comment:
Are there known types?

Reply:
Here the type is "nutrition_extractor" do we have an object for this?

Reply:
> Here the type is "nutrition_extractor" do we have an object for this?

I guess you'll have to create a new class or enum for that.

Reply:
I'm not sure what level of abstraction this represents. Is it okay to leave it as a string until we have a second endpoint at the same level?

Reply:
If you have a limited understanding of what this field is for, and as a consequence about its usage, it would be better to ignore that field temporarily (e.g. commenting and adding a `TODO`).
Once we've decided that a field has a given type, it's a breaking change to change its type. Let's avoid that.

Reply:
Good point, updating now